codex-process-data-transfer/src/main/java/de/netzwerk_universitaetsmedizin/codex/processes/data_transfer/service/send/ReadData.java

```diff
@@ -741,10 +768,18 @@ else if (resource instanceof Procedure p)
 				p.setIdentifier(Collections.emptyList());
 				p.setSubject(patientRef);
 			}
+			else if (resource instanceof Specimen s)
+			{
+				s.setIdentifier(Collections.emptyList());
+				s.setAccessionIdentifier(null);
+				s.setSubject(patientRef);
+			}
 			else
 				throw new RuntimeException("Resource of type " + resource.getResourceType().name() + " not supported");
 		}
 
+		((DomainResource) resource).getContained().forEach(r -> setSubjectOrIdentifier(r, pseudonym));
```

Comment:
This cast should be guarded, Binary and Bundle are not DomainResources and may potentially be transferred in the future.
```suggestion
		if (resource instanceof DomainResource d)
			d.getContained().forEach(r -> setSubjectOrIdentifier(r, pseudonym));
		else
			throw new RuntimeException("Resource of type " + resource.getResourceType().name() + " not supported");
```

Reply:
You are correct, my IDE just tells me that this check is always true, so I removed it. Will introduce it again.